bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/DynamicChannelHelper.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.airgradient.internal.handler;
+
+import static org.openhab.binding.airgradient.internal.AirGradientBindingConstants.BINDING_ID;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DynamicChannelHelper} is responsible for creating dynamic configuration channels.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class DynamicChannelHelper {
+
+    private record ConfigurationChannel(String id, String typeId, String itemType) {
+    }
+
+    private final static List<ConfigurationChannel> channels = new ArrayList<ConfigurationChannel>() {
+        {
+            add(new ConfigurationChannel("country-code", "country-code", "String"));
```

Comment:
Instead of using these strings, you can re-use the binding constants. e.g. `CHANNEL_COUNTRY_CODE`

Reply:
Off course, why didnt I think about that. I kept channelid and typeid as two different concepts even though the values are the same in case there comes something configurable that lets us reuse types in the future.